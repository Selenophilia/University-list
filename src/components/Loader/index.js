import React from "react";
import "./index.scss";

const Loader = ({ ...props }) => {
  return (
    <div data-testid="loader" className="container">
      <i type="icon-spinner animate-spin" className="loading" {...props} />
    </div>
  );
};

export default Loader;
