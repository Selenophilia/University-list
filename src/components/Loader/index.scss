.container {
  display: flex;
  align-items: center;
  justify-content: center;
  .loading {
    padding: 8px;
    border: 3px solid "red";
    border-right-color: "red";
    border-radius: 50%;
    transition: 0.2s all ease-in;
    position: absolute;
    animation: rotate 1s infinite linear;
  }

  @keyframes rotate {
    100% {
      -webkit-transform: rotate(360deg);
    }
  }
}
