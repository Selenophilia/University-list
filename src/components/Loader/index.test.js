import React from "react";
import { render } from "@testing-library/react";
import Loader from ".";

const renderRoot = () => {
  return render(<Loader />);
};

it("renders Loader component correctly ", () => {
  const { asFragment } = renderRoot();
  expect(asFragment()).toMatchSnapshot();
});
