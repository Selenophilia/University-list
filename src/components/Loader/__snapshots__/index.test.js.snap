// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loader component correctly  1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="loader"
  >
    <i
      class="loading"
      type="icon-spinner animate-spin"
    />
  </div>
</DocumentFragment>
`;
